modules/apps/portal-language/portal-language-test/src/testIntegration/java/com/liferay/portal/language/test/LanguageResourcesTest.java

```diff
@@ -120,19 +134,64 @@ public void testLanguageResourceServiceTrackerCustomizerServiceRanking() {
 		_assertValue(null);
 	}
 
+	@Test
+	public void testLanguageResourceUsingSupportedLocale() {
+		String key = "year";
+
+		String expectedTranslation = "Année";
+
+		Locale locale = LocaleUtil.FRANCE;
+
+		Assert.assertEquals(
+			expectedTranslation, _language.get(locale, key, null));
+
+		_serviceRegistration1 = _register(
+			_VALUE_1, 0, _language.getLanguageId(locale));
+
+		_assertValue(_VALUE_1, locale);
+	}
+
+	@Test
+	public void testLanguageResourceUsingUnsupportedLocale() {
+		String key = "year";
+
+		String expectedBaseLanguagePropertiesTranslation = "Year";
+
+		Locale locale = new Locale("ps", "AF");
+
+		Assert.assertEquals(
+			expectedBaseLanguagePropertiesTranslation,
+			_language.get(locale, key, null));
+
+		_serviceRegistration1 = _register(
+			_VALUE_1, 0, _language.getLanguageId(locale));
+
+		_assertValue(_VALUE_1, locale);
```

Comment:
@wanderlast This part was used to test service tracker, why you need to test it in your context?